crates/forge/bin/cmd/create.rs

```diff
@@ -157,123 +157,135 @@ impl CreateArgs {
+                                    let fdep_art = zk_output
+                                        .find(abs_path_str, contract_name)
+                                        .unwrap_or_else(|| {
+                                            panic!(
                                     "Could not find contract {} at path {} for compilation output",
                                     contract_name, contract_path)
-                                    });
-                                let fdep_fdeps_map = fdep_art
-                                    .factory_dependencies
-                                    .clone()
-                                    .expect("factory deps not found");
-                                for dep in fdep_fdeps_map.values() {
-                                    queue.push_back(dep.clone())
+                                        });
+                                    let fdep_fdeps_map = fdep_art
+                                        .factory_dependencies
+                                        .clone()
+                                        .expect("factory deps not found");
+                                    for dep in fdep_fdeps_map.values() {
+                                        queue.push_back(dep.clone())
+                                    }
+
+                                    let fdep_bytecode = fdep_art
+                                        .bytecode
+                                        .clone()
+                                        .expect("Bytecode not found for factory dependency")
+                                        .object
+                                        .clone()
+                                        .into_bytes()
+                                        .unwrap()
+                                        .to_vec();
+                                    visited_bytecodes.insert(fdep_bytecode);
                                 }
-
-                                let fdep_bytecode = fdep_art
-                                    .bytecode
-                                    .clone()
-                                    .expect("Bytecode not found for factory dependency")
-                                    .object
-                                    .clone()
-                                    .into_bytes()
-                                    .unwrap()
-                                    .to_vec();
-                                visited_bytecodes.insert(fdep_bytecode);
                             }
+
+                            visited_bytecodes.insert(bytecode.clone());
+                            let factory_deps: Vec<Vec<u8>> =
+                                visited_bytecodes.into_iter().collect();
+                            let zk_data = ZkSyncData { bytecode, bytecode_hash, factory_deps };
+                            (abi, bin.object, Some(zk_data))
```

Comment:
Can we reduce this nesting somehow? It's a bit hard to follow the logic of option matching

Reply:
I think we should re-extract this code into a seprate function (to lookup factory deps) since it does make it a bit longer